crates/astria-sequencer/src/grpc/storage/values/sequencer_block_header.rs

```diff
@@ -0,0 +1,182 @@
+        let timestamp = tendermint::time::Time::from_unix_timestamp(
+            i64::try_from(nanos / 1_000_000_000).unwrap(),
+            (nanos % 1_000_000_000) as u32,
+        )
+        .map_err(std::io::Error::other)?;
+        Ok(BlockTimestamp(timestamp))
+    }
+}
+
+#[derive(Debug, BorshSerialize, BorshDeserialize)]
+pub(in crate::grpc) struct SequencerBlockHeader<'a> {
+    chain_id: ChainId<'a>,
+    height: u64,
+    time: BlockTimestamp,
+    rollup_transactions_root: Cow<'a, [u8; 32]>,
+    data_hash: Cow<'a, [u8; 32]>,
+    proposer_address: [u8; ADDRESS_LEN],
+}
+
+impl<'a> SequencerBlockHeader<'a> {
+    pub(super) fn height(&self) -> u64 {
+        self.height
+    }
+}
+
+impl<'a> From<&'a DomainSequencerBlockHeader> for SequencerBlockHeader<'a> {
+    fn from(header: &'a DomainSequencerBlockHeader) -> Self {
+        const _: () = assert!(ADDRESS_LEN == tendermint::account::LENGTH);
+        let mut proposer_address = [0; ADDRESS_LEN];
+        proposer_address.copy_from_slice(header.proposer_address().as_bytes());
+
+        SequencerBlockHeader {
+            chain_id: header.chain_id().into(),
+            height: header.height().value(),
+            time: header.time().into(),
+            rollup_transactions_root: Cow::Borrowed(header.rollup_transactions_root()),
+            data_hash: Cow::Borrowed(header.data_hash()),
+            proposer_address,
+        }
+    }
+}
+
+impl<'a> From<SequencerBlockHeader<'a>> for DomainSequencerBlockHeader {
+    fn from(header: SequencerBlockHeader<'a>) -> Self {
+        let height = tendermint::block::Height::try_from(header.height)
+            .expect("should not be able to store invalid height");
+        DomainSequencerBlockHeader::unchecked_from_parts(SequencerBlockHeaderParts {
+            chain_id: header.chain_id.into(),
+            height,
+            time: header.time.into(),
+            rollup_transactions_root: header.rollup_transactions_root.into_owned(),
+            data_hash: header.data_hash.into_owned(),
+            proposer_address: tendermint::account::Id::new(header.proposer_address),
+        })
+    }
+}
+
+impl<'a> From<SequencerBlockHeader<'a>> for crate::storage::StoredValue<'a> {
+    fn from(block_header: SequencerBlockHeader<'a>) -> Self {
+        crate::storage::StoredValue::Grpc(Value(ValueImpl::SequencerBlockHeader(block_header)))
+    }
+}
+
+impl<'a> TryFrom<crate::storage::StoredValue<'a>> for SequencerBlockHeader<'a> {
+    type Error = astria_eyre::eyre::Error;
+
+    fn try_from(value: crate::storage::StoredValue<'a>) -> Result<Self, Self::Error> {
+        let crate::storage::StoredValue::Grpc(Value(ValueImpl::SequencerBlockHeader(block_header))) =
+            value
+        else {
+            bail!(
+                "grpc stored value type mismatch: expected sequencer block header, found {value}"
+            );
+        };
+        Ok(block_header)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::str::FromStr;
```

Comment:
Just `parse` in the test instead of importing?

Reply:
Done in [dd3bfa6](https://github.com/astriaorg/astria/pull/1492/commits/dd3bfa68494d3d14555a09c0dfb5cbe6228b6f7d).